config-service-impl/src/main/java/org/hypertrace/config/service/store/DocumentConfigStore.java

```diff
@@ -180,4 +181,15 @@ private ContextSpecificConfig getContextSpecificConfig(ConfigDocument configDocu
         .setUpdateTimestamp(configDocument.getUpdateTimestamp())
         .build();
   }
+
+  private InternalContextSpecificConfig getInternalContextSpecificConfig(
+      ConfigDocument configDocument, ContextSpecificConfig existingConfig) {
+    if (ConfigServiceUtils.isNull(existingConfig.getConfig())) {
+      return new InternalContextSpecificConfig(
+          getContextSpecificConfig(configDocument), Optional.empty());
+    } else {
+      return new InternalContextSpecificConfig(
```

Comment:
nit- cleaner to overload constructor rather than accept an optional (optional params could technically be null, which makes it better suited as a return type than a param type).